.github/workflows/publish.yml

```diff
@@ -1,21 +1,52 @@
-name: publish
+name: Build and Publish Docker Image
 
 on:
   push:
-    branches: [master, develop]
+    branches:
+        - "master"
+        - "develop"
+    tags:
+      - "v*.*.*"
```

Comment:
This section ensures that this workflow will only execute when:
- a pull request is merged into `master` or `develop`
- a new git tag event occurs AND the git tag complies with semantic versioning syntax